fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/KeySpacePath.java

```diff
@@ -615,4 +615,14 @@ default void deleteAllData(@Nonnull FDBRecordContext context) {
     default List<KeySpacePath> list(@Nonnull FDBRecordContext context, @Nonnull String subdirName) {
         return list(context, subdirName, ScanProperties.FORWARD_SCAN);
     }
+
+    /**
+     * String representation of this {@link KeySpacePath} that shows correspondences between original and resolved
+     * directory values in accordance with the input Tuple.
+     *
+     * @param tuple representing the directory values of this path in resolved form
+     *
+     * @return a string representation of the path that shows both original and resolved directory values
+     */
+    String toString(@Nonnull Tuple tuple);
```

Comment:
Being a little forward looking, I think that this method, and its implementation should be moved into `KeySpacePathSubspaceProvider`.  I have a PR out right now (https://github.com/FoundationDB/fdb-record-layer/pull/414) that will give us a cleaner mechanism (IMHO) for carrying around a path with its tuple elements. If you put the logic inside of the provider and private methods, then we we can switch out implementations at a later date rather easily.   If there is an argument for keeping it here, I would suggest that we mark it as `UNSTABLE` or `INTERNAL` or something like that.

Reply:
Marked the API as UNSTABLE for now. We can move it if necessary with the future work.